js/alert.js

```diff
@@ -0,0 +1,177 @@
+let IsGettingAlertsucceed = false;
+
+function alertHandler() {
+  getAlert();
+  setAlertTable(IsGettingAlertsucceed);
+  setAlertInterval(IsGettingAlertsucceed);
+}
+
+function getAlert() {
+  const alertRateValue = Number(alertRate.value);
+
+  // change IsGettingAlertsucceed value to false if it is true from previous time
+  IsGettingAlertsucceed = false;
+
+  let isItNewAlert = true;
+  let alertIndex;
+  alerts.forEach((alertItem, index) => {
+    if (
+      alertItem.base === alertBaseList.value &&
+      alertItem.target === alertTargetList.value
+    ) {
+      isItNewAlert = false;
+      alertIndex = index;
+    }
+  });
+
+  if (!alertBaseList.value) {
+    showErrorMessage(alertBaseList, alertError, "Select A Currency Please");
+  } else if (!alertTargetList.value) {
+    showErrorMessage(alertTargetList, alertError, "Select A Currency Please");
+  } else if (alertRateValue <= 0) {
+    showErrorMessage(alertRate, alertError, "Invalid Input");
+  } else if (!isItNewAlert) {
+    setAlertYesNoQuestionBox(
+      alertIndex,
+      alertRateValue,
+      "An Alert Has Already Been Set For This Rate. Would You Like To Update It?!"
+    );
+  } else {
+    alerts.push({
+      id: `${alertBaseList.value}-${alertTargetList.value}`,
+      base: alertBaseList.value,
+      target: alertTargetList.value,
+      rate: alertRateValue,
+    });
+    IsGettingAlertsucceed = true;
+
+    showSuccessMessage(" Alert Set !!");
+
+    alertBaseList.value = "";
+    alertTargetList.value = "";
+    alertRate.value = "";
+  }
+}
+
+function setAlertInterval(isValid) {
```

Comment:
Instead of passing `isValid` as a parameter and wrapping all your function in a condition, you could simply not call the `setAlertInterval` function:

```js
if (isGettingAlertSucceed) {
  setAlertInterval();
}
```